binding/ios/Leopard.swift

```diff
@@ -0,0 +1,172 @@
+//  You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+//  file accompanying this source.
+//  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+//  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+//  specific language governing permissions and limitations under the License.
+//
+
+import PvLeopard
+
+/// iOS binding for Leopard wake word engine. Provides a Swift interface to the Leopard library.
+public class Leopard {
+    private static let supportedAudioTypes: Set = ["flac", "mp3", "ogg", "opus", "wav", "webm"]
+
+    private var handle: OpaquePointer?
+    public static let sampleRate = UInt32(pv_sample_rate())
+    public static let version = String(cString: pv_leopard_version())
+
+    /// Constructor.
+    ///
+    /// - Parameters:
+    ///   - accessKey: The AccessKey obtained from Picovoice Console (https://console.picovoice.ai).
+    ///   - modelPath: Absolute path to file containing model parameters.
+    /// - Throws: LeopardError
+    public init(accessKey: String, modelPath: String) throws {
+
+        if accessKey.count == 0 {
+            throw LeopardInvalidArgumentError("AccessKey is required for Leopard initialization")
+        }
+
+        if !FileManager().fileExists(atPath: modelPath) {
+            throw LeopardInvalidArgumentError("Model file at does not exist at '\(modelPath)'")
+        }
+
+        let status = pv_leopard_init(
+                accessKey,
+                modelPath,
+                &handle)
+        try checkStatus(status, "Leopard init failed")
+    }
+
+    /// Constructor.
+    ///
+    /// - Parameters:
+    ///   - accessKey: The AccessKey obtained from Picovoice Console (https://console.picovoice.ai).
+    ///   - modelURL: URL file containing model parameters.
+    /// - Throws: LeopardError
+    public convenience init(accessKey: String, modelURL: URL) throws {
+        try self.init(accessKey: accessKey, modelPath: modelURL.path)
+    }
+
+    deinit {
+        self.delete()
+    }
+
+    /// Releases native resources that were allocated to Leopard
+    public func delete(){
+        if handle != nil {
+            pv_leopard_delete(handle)
+            handle = nil
+        }
+    }
+
+    /// Processes a given audio data and returns its transcription.
+    ///
+    /// - Parameters:
+    ///   - pcm: An array of 16-bit pcm samples. The audio needs to have a sample rate equal to `.sample_rate` and be 16-bit
+    ///          linearly-encoded. This function operates on single-channel audio. If you wish to process data in a different
+    ///          sample rate or format consider using `.process_file`.
+    /// - Throws: LeopardError
+    /// - Returns: Inferred transcription.
+    public func process(_ pcm:[Int16]) throws -> String {
+        if handle == nil {
```

Comment:
is this even possible? just figuring if we need this check

Reply:
yes, if you delete and run process.